core/consensus/slots/src/lib.rs

```diff
@@ -207,32 +209,43 @@ pub enum CheckedHeader<H, S> {
 	Checked(H, S),
 }
 
+/// A type from which a slot duration can be obtained.
+pub trait SlotData {
+	/// Gets the slot duration.
+	fn slot_duration(&self) -> u64;
```

Comment:
I don't think this will work for integration in runtimes and I don't know of a strong case for adding yet another runtime API for consensus support. Better to have it in the main BABE trait.

Reply:
@rphmeier This code is present for the benefit of the slots crate, but I could just use a closure instead.

Reply:
The trait very much helps the slots code, and leads to a cleaner API there, but the runtime doesn’t need to know about it, only the [`BabeApi`](/paritytech/substrate/blob/0eb29f6fce610c6c8f79e1e585d9c4221148f6f5/core/consensus/babe/primitives/src/lib.rs#L62-L71) trait.